test/algorithms/physical_design/graph_oriented_layout_design.cpp

```diff
@@ -0,0 +1,187 @@
+#include <fiction/layouts/cartesian_layout.hpp>
+#include <fiction/layouts/cell_level_layout.hpp>
+#include <fiction/layouts/clocked_layout.hpp>
+#include <fiction/layouts/gate_level_layout.hpp>
+#include <fiction/layouts/tile_based_layout.hpp>
+#include <fiction/networks/technology_network.hpp>
+#include <fiction/technology/qca_one_library.hpp>
+#include <fiction/utils/network_utils.hpp>
+
+#include <mockturtle/networks/aig.hpp>
+#include <mockturtle/networks/mig.hpp>
+#include <mockturtle/views/fanout_view.hpp>
+#include <mockturtle/views/names_view.hpp>
+
+#include <type_traits>
+
+using namespace fiction;
+
+template <typename Lyt, typename Ntk>
+void check_graph_oriented_layout_design_equiv(const Ntk& ntk)
+{
+    graph_oriented_layout_design_stats  stats{};
+    graph_oriented_layout_design_params params{};
+    params.timeout      = 100000;
+    params.return_first = true;
+
+    const auto layout = graph_oriented_layout_design<Lyt>(ntk, params, &stats);
+    REQUIRE(layout.has_value());
+
+    check_eq(ntk, *layout);
+}
+
+template <typename Lyt>
+void check_graph_oriented_layout_design_equiv_all()
+{
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::maj1_network<mockturtle::aig_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::maj4_network<mockturtle::aig_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::unbalanced_and_inv_network<mockturtle::aig_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::and_or_network<technology_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::half_adder_network<mockturtle::mig_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::full_adder_network<technology_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::mux21_network<mockturtle::xag_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::se_coloring_corner_case_network<technology_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(
+        blueprints::fanout_substitution_corner_case_network<technology_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::inverter_network<technology_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::clpl<technology_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(
+        blueprints::one_to_five_path_difference_network<technology_network>());
+    check_graph_oriented_layout_design_equiv<Lyt>(blueprints::nand_xnor_network<technology_network>());
+}
+
+TEST_CASE("Layout equivalence", "[algorithms]")
+{
+    SECTION("Cartesian layouts")
+    {
+        using gate_layout = gate_level_layout<clocked_layout<tile_based_layout<cartesian_layout<offset::ucoord_t>>>>;
+
+        check_graph_oriented_layout_design_equiv_all<gate_layout>();
+    }
+}
+
+TEST_CASE("Gate library application", "[graph_oriented_layout_design]")
```

Comment:
Replace the tag with `graph-oriented-layout-design` (hyphens instead of underscores) everywhere. Also make sure that the test names are unique across the entire test suite. Otherwise, it could lead to issues on some systems assuming that test names are overloaded.